### PR TITLE
Subgraph and merge

### PR DESCRIPTION
**Pull request by oortsang**
I think the tests should be ok, but note I've put the subgraph/permute/subgraph_relabel functions in `test_api_enhance.py` (moved from `..._enchance.py`) since that's where the `permute` tests were originally. Merge tests are in `test_util_merges`.

### CODE REVIEW

File: tinygraph/util.py [hunk 1]
```diff
@@ -33,29 +32,189 @@ def graph_equality(g1, g2):
...
 
     return True
-    
+
+def subgraph_relabel(g, node_iter):
+    """
+    Helper function to perform the work of permute and subgraph.
+    Formally, it returns the subgraph of g induced by node_iter (as the set of vertices)
+    Maintains the ordering of node_iter in constructing the new subgraph.
+
```
[the remark this answers is not in the document]
Comment (oortsang) [reply tinygraph/util.py:41]: Ok, should I rename the function to `_subgraph_relabel` in that case?

[the remark this answers is not in the document]
Comment (ericmjonas) [reply tinygraph/util.py:41]: Yeah I think that's probably a good idea?

File: tinygraph/util.py [hunk 2]
```diff
@@ -33,29 +32,189 @@ def graph_equality(g1, g2):
...
+    if N == 0:
+        # Weird things happen if we keep going ;_;
+        return new_g
+
+    # Magic index converter eliminates python-for-loops
+    # at the expense of ignoring sparsity (moves everything)
+    new_list = np.arange(N) # list of the new vertices
+    # node_iter is a list of old vertices ordered by destination indices
+    ii = (np.repeat(new_list,  N),  np.tile(new_list,  N))
```
[the remark this answers is not in the document]
Comment (oortsang) [reply tinygraph/util.py:67]: Got it! Thanks!

File: tests/test_api_enhance.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    pG11 = permute(g1,   [3,4,1,2,0])
+    pG12 = permute(g1,   [4,3,1,2,0])
+    pG13 = permute(pG11, [4,2,3,0,1])
+
+    assert not graph_equality(g2, pG11)
+    assert graph_equality(g2, pG12)
+    assert graph_equality(g1, pG13)
+
+def test_permute_path():
+    """Another case"""
```
[the remark this answers is not in the document]
Comment (oortsang) [reply tests/test_api_enhance.py:125]: Oops, good catch!

File: tinygraph/util.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (oortsang) [reply tinygraph/util.py:41]: Cool, I'll do that then.

**Comment by oortsang** [comment]
Oh haha, today I learned that wrapping text in backticks in git commit messages does not get treated as plaintext (`subgraph_relabel: command not found` and `_subgraph_relabel: command not found`). So that is why the commit message just says "[nothing] renamed to [nothing];".

**Review by ericmjonas — APPROVED** [review]

**Comment by ericmjonas** [comment]
@oortsang this looks great go ahead and merge

**Pull request merged**